ts/nni_manager/training_service/reusable/environments/heterogenousEnvironmentService.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import { EventEmitter } from "events";
+import * as fs from 'fs';
+import * as path from 'path';
+import * as component from '../../../common/component';
+import { getLogger, Logger } from '../../../common/log';
+import { TrialConfigMetadataKey } from '../../common/trialConfigMetadataKey';
+import { HeterogenousCommandChannel } from '../channels/heterogenousCommandChannel';
+import { CommandChannel } from "../commandChannel";
+import { EnvironmentInformation, EnvironmentService } from '../environment';
+import { AMLEnvironmentService } from './amlEnvironmentService';
+import { RemoteEnvironmentService } from './remoteEnvironmentService';
+import { randomSelect } from '../../../common/utils';
+
+
+/**
+ * Collector PAI jobs info from PAI cluster, and update pai job status locally
+ */
+@component.Singleton
+export class HeteroGenousEnvironmentService extends EnvironmentService {
+    
+    private amlEnvironmentService: AMLEnvironmentService;
+    private remoteEnvironmentService: RemoteEnvironmentService;
+
+    private readonly log: Logger = getLogger();
+
+    constructor() {
+        super();
+        this.amlEnvironmentService = new AMLEnvironmentService();
```

Comment:
It can be more flexible, like to support two aml instances. So the list should be created dynamically. But we need some assumption about order of config values or other places to make sure we count it correctly.

Reply:
EnvironmentService is singleton, does not have two instances.

Reply:
How to support two aml instances? The user may have such a requirement. I wanted to use multiple OpenPAI clusters before, but I have to use only one, since NNI doesn't support it.

Reply:
Refactored code. In current code logic, we still use `heterogeneousConfig` to set heterogeneous mode internally, @liuzhe will add a convertor to convert this config to new yaml format for users. In this `heterogeneousConfig` version, we could only support one aml instance, multiple instances will be supported in next release. Next release, we will refactor config setting logic, and refactor setClusterMetadata and create instance logic here.